tests/beaker_tests/bgp/bgplib.rb

```diff
@@ -34,22 +34,36 @@ module BgpLib
   VRF2        = 'red'
 
   # Create manifest ensure => present + 'default' property values
-  def self.create_bgp_manifest_present
+  def self.create_bgp_manifest_present(platform, vrf='default')
+    conditional_props = ''
+    if platform != 'ios_xr'
+      conditional_props <<
+        "maxas_limit                            => 'default',
+         shutdown                               => 'default',
+         suppress_fib_pending                   => 'default',
+         bestpath_med_non_deterministic         => 'default',
+         timer_bestpath_limit                   => 'default',
+         timer_bestpath_limit_always            => 'default',
+         graceful_restart_helper                => 'default',
+         "
+    end
+
+    if vrf != 'default'
+      conditional_props <<
+        "route_distinguisher                    => 'default',"
+    end
+
     manifest_str = "cat <<EOF >#{UtilityLib::PUPPETMASTER_MANIFESTPATH}
     node 'default' {
       cisco_bgp { 'default':
         ensure                                 => present,
         asn                                    => #{BgpLib::ASN},
-        vrf                                    => 'default',
+        vrf                                    => '#{vrf}',
```

Comment:
Seems a little out of place - because the whole example is defaulted. But as the default=default(sorry) I guess it's ok.